### ISSUE TITLE
Data Scientists Should Know Software Engineering Best Practices

### ISSUE THREAD

**Issue by jessonpagaduan** [khuyentran1401/machine-learning-articles#90]
## TL;DR
To be an indispensable data scientist, one should know and apply software engineering best practices.

### Article Link
https://towardsdatascience.com/data-scientist-should-know-software-engineering-best-practices-f964ec44cada

### Author
Kurtis Pykes

## Key Takeaways
Software engineering practices data scientists should know:
1. Clean code (adhere to PEP8 for Pythonistas);
2. Applying effective modularity in coding (e.g. don't repeat yourself, single responsibility principle, open-closed principle)
3. Code refactoring for improved performance and readability and reduced complexity
4. Testing
5. Code review

## Useful Code Snippets
N/A

## Useful Tools
PEP8

## Comments/ Questions
I must adhere to PEP8 consistently when I code in Python.

**Comment by khuyentran1401** [comment]
I read this article! This is a really good article on good software engineering practices for data scientists!
